### PR TITLE
Implement rectangle.contains? in rust

### PR DESCRIPTION
**Pull request by rwstauner**
Thanks for a great presentation!
and for getting me to finally write my first rust code :-)

```
GeoTest
  test_rectangle_contains_true                                    PASS (0.00s)
  test_point_distance                                             PASS (0.00s)
  test_rectangle_contains_false                                   PASS (0.00s)
```

see also https://github.com/rwstauner/2022-09-09-ruby-on-rust-intro/pull/1 🤦

### CODE REVIEW

**Comment by ianks** [comment]
Great work! Thanks for contributing this and hope you get a chance to dive into more Rust!

File: demos/ext/rust_magnus_geo/src/lib.rs [hunk 1]
```diff
@@ -77,11 +77,14 @@ impl Rectangle {
...
     /// Checks if the given `Point` is inside this `Rectangle`.
-    fn contains(&self, _other: &Point) -> Result<bool, Error> {
-        let _top_left = self.top_left.try_convert::<&Point>()?;
-        let _bottom_right = self.bottom_right.try_convert::<&Point>()?;
+    fn contains(&self, other: &Point) -> Result<bool, Error> {
+        let top_left = self.top_left.try_convert::<&Point>()?;
+        let bottom_right = self.bottom_right.try_convert::<&Point>()?;
 
-        Err(Error::new(not_imp_error(), "finish me!"))
+        Ok(   other.x >= top_left.x
```
Comment (ianks) [inline demos/ext/rust_magnus_geo/src/lib.rs:84]: Small tip: check out the rust-analyzer vscode extension. It will setup auto-formatting for ya

Comment (rwstauner) [reply demos/ext/rust_magnus_geo/src/lib.rs:84]: i used vim, and apparently my setup (probably ale) brings a free linter that removed the unused `not_imp_error` but i don't have auto-formatting setup

**Comment by rwstauner** [comment]
closing to remove from pr lists.  thanks for this!

**Pull request closed without merging**